### PR TITLE
[DON'T MERGE] update to sqlv1beta1 gateway client and drop identityPool support

### PR DESCRIPTION
**Pull request by YannickPferr**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * no: DO NOT MERGE until the required features are live in prod  

What
----

- adds support for using service accounts. When users want to run an interactive query with just their user identity, we display a warning, to let them know their statements will timeout after 4h.
- switches to v1beta1 SDK which supports service account in body

Merge after: #2210 and after https://github.com/confluentinc/cc-flink-gateway-service-v2/pull/326

### CODE REVIEW

File: pkg/ccloudv2/flink_gateway.go
Comment (brianstrauch) [inline pkg/ccloudv2/flink_gateway.go:90]: Can we create `statementObj` outside of this function? The long-term goal is for these functions to be auto-generated. They shouldn't contain any extra logic.

File: pkg/flink/app/application.go
Comment (brianstrauch) [inline pkg/flink/app/application.go:126]:
```suggestion
	if err := a.refreshToken(); err != nil {
```

File: pkg/flink/internal/controller/statement_controller.go
Comment (brianstrauch) [inline pkg/flink/internal/controller/statement_controller.go:45]: Why `utils.OutputWarnf()` and not `log.CliLogger.Warnf()`? This is what all other commands use.

File: pkg/flink/internal/utils/output.go
Comment (brianstrauch) [inline pkg/flink/internal/utils/output.go:25]: I wouldn't expect any function ending in `f` to add a newline.

File: pkg/flink/internal/controller/statement_controller.go
Comment (YannickPferr) [reply pkg/flink/internal/controller/statement_controller.go:45]: These functions colorize the output message (warn messages will be yellow for example)

**Comment by Guttz** [comment]
❗ Requested product changes before merging: https://confluent.slack.com/archives/C04S2SKUKEV/p1693268040317669?thread_ts=1692990849.924799&cid=C04S2SKUKEV

File: pkg/flink/internal/controller/statement_controller.go
Comment (luca-filipponi) [reply pkg/flink/internal/controller/statement_controller.go]: done

File: pkg/flink/internal/utils/output.go
Comment (luca-filipponi) [reply pkg/flink/internal/utils/output.go]: removed from here and added to the caller

File: pkg/ccloudv2/flink_gateway.go
Comment (luca-filipponi) [reply pkg/ccloudv2/flink_gateway.go]: done

File: pkg/flink/app/application.go
Comment (luca-filipponi) [reply pkg/flink/app/application.go]: done

File: pkg/flink/internal/controller/statement_controller.go
Comment (brianstrauch) [inline pkg/flink/internal/controller/statement_controller.go:45]:
```suggestion
		utils.OutputWarnf(`[WARN] To ensure that your statements run continuously, switch to using a service account instead of your user identity by running "SET '%s'='sa-123';". Otherwise, statements will stop running after 4 hours.`,
```

File: pkg/flink/internal/store/store.go
Comment (brianstrauch) [inline pkg/flink/internal/store/store.go:7]: `make lint`?

Comment (luca-filipponi) [reply pkg/flink/internal/store/store.go]: done ✅

**Review by luca-filipponi — APPROVED** [review]

**Comment by luca-filipponi** [comment]
Support for SA is not in prod yet, so we can't merge this as it will remove support for identityPools

**Pull request closed without merging**
